### DISCUSSION TITLE
CodeQL - Low on disk space error

### DISCUSSION THREAD

**Discussion by SelwynC** [github/securitylab#736] [Q&A]
I am using CodeQL and since the project is large I am running low on the disk space.
Can we clean up the disk space before the analysis can you share any sample snippet on achieving this.
![image](https://user-images.githubusercontent.com/117057125/224878389-6390c6fa-900d-4dc8-80f4-6d95dda5f258.png)

**Comment by smowton** [comment]
In which phase does the action run out of space? (This affects what temporary artefacts could be removed)

Does an action freeing up space at the start such as https://github.com/marketplace/actions/maximize-build-disk-space help?

**Reply by SelwynC** [reply]
This takes place during the auto build step and I am using windows-latest runner does it matter ?
Please see the script below, 
![image](https://user-images.githubusercontent.com/117057125/225224678-9ae9c4fa-abb6-4451-93ff-056f1ee0ddff.png)

**Reply by smowton** [reply]
In that case I'm afraid you need to either clean up disk space before the autobuild phase starts as noted above, or build a more restricted subset of your code for analysis by replacing the autobuild step with a manual build command. Enterprise users can also use Actions runners with more resources (https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners)
